projects/js-packages/partner-coupon/hooks.js

```diff
@@ -48,11 +48,13 @@ export function usePartnerCouponRedemption(
 			} );
 		}
 
-		window.location.href = getRedirectUrl( 'jetpack-plugin-partner-coupon-checkout', {
-			path: partnerCoupon.product.slug,
-			site: siteRawUrl,
-			query: `coupon=${ partnerCoupon.coupon_code }`,
-		} );
+		window.location.assign(
+			getRedirectUrl( 'jetpack-plugin-partner-coupon-checkout', {
+				path: partnerCoupon.product.slug,
+				site: siteRawUrl,
+				query: `coupon=${ partnerCoupon.coupon_code }`,
+			} )
+		);
```

Comment:
I've chosen to swap to `window.location.assign` here because it made more sense in terms of testing.
Practically speaking, it achieves the same thing for the user.